browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -185,10 +185,19 @@ void BraveNewTabPageHandler::OnSearchPromotionDismissed() {
   NotifySearchPromotionDisabledIfNeeded();
 }
 
-void BraveNewTabPageHandler::UseColorBackground(const std::string& color) {
+void BraveNewTabPageHandler::UseColorBackground(const std::string& color,
+                                                bool use_random_color) {
+  if (use_random_color) {
+    CHECK(color == brave_new_tab_page::mojom::kRandomSolidColorValue ||
```

Comment:
IMO, this should be `DCHECK`.
It doesn't make sence to me to make official browser crashed with this kind of check failure. WDYT?
If this happenes in release build, I think NTP will just use the `color`?

Reply:
Yeah, we can change this to DCHECK. When this happens, it's likely to be undefined behavior. I guess NTP will use random gradient value as a result of `if/else` condition.

handwavingly..
```
if (useRandomColor)
  return |color| == RANDOM_SOLID_COLOR  ? PickOneFromSolidColors() : PickOneFromGradientColors()
```